### PR TITLE
Adds Messages() function

### PR DESCRIPTION
**Pull request by ebh**
This PR probably should not be merged as-is :)

@elliotchance, I could see several different ways to implement, none I was 100% sure were the best way. So I've raised this one, which I know is not perfect, to give us a starting point for you to decide what is the best way.

I'll also cover one of the ways considered but dismissed and why.

I thought of just having:
```
func (queue *Queue) Messages() *orderedmap.OrderedMap {
	return queue.messages
}
```

But as this would return a pointer to the internal state of the `Queue` I thought that was unsafe. A consumer could inadvertently change the internal state of the `Queue`.

So I went with the implementation you see here. Where I create a new `OrderedMap`, copy the messages, and return a pointer to the new `OrderedMap`.

The questions I have with the approach I've taken are:
1. Should I be returning an `OrderedMap` or `[]interface{}`, only containing the values?
2. Should I be setting a clearing a lock? And if so how?
3. How do I test the `if !ok {...}`?

---
This change is [Reviewable](https://reviewable.io/reviews/elliotchance/mocksqs/10)

### CODE REVIEW

File: queue.go
Comment (elliotchance) [inline queue.go]: This is a totally reasonable approach, with one modification:

`Messages` should ideally block write access to the queue. This will gaurantee under a concurrent environment that a true snapshot is taken, but probably more importantly you don't need to check `Get` since all keys will exist while write access is blocked. However, to do this you would need to create a `sync.RWMutex` on the `Queue`, then add the correct locks to each method (not just `Messages`).

This also has the added bonus that orderedmap itself is not thread-safe so the current implementation of mocksqs could actually mangle the queue:

```go
type Queue struct {
 	sync.RWMutex
 	// ...
}

func (queue *Queue) Messages() *orderedmap.OrderedMap {
 	queue.Lock()
 	defer queue.Unlock()

 	rv := orderedmap.NewOrderedMap()
 	for el := m.Front(); el != nil; el = el.Next() {
 		rv.Set(el.Key, el.Value)
 	}

 	return rv
}
```

For bonus points, you could create a PR for orderedmap to provide a `Copy` method.

**Review by elliotchance — CHANGES_REQUESTED** [review]

File: queue.go
Comment (ebh) [reply queue.go]: Thanks. That is a very clear and helpful response 🙏 😄 

I've decided to go for the bonus points and add a `Copy()` function (see [PR #19](https://github.com/elliotchance/orderedmap/pull/19)).

**Review by elliotchance — APPROVED** [review]

**Comment by elliotchance** [comment]
Excellent work, thank you Alex. However, I cannot merge it until the merge conflicts are resolved.

**Comment by ebh** [comment]
Sorry, didn't realise there were merge conflicts. They have now been resolved

**Review by elliotchance — APPROVED** [review]

**Pull request merged**
